app/helpers/form_helper.rb

```diff
@@ -202,22 +202,16 @@ def file_field_f(f, attr, options = {})
   end
 
   def autocomplete_f(f, attr, options = {})
-    url = options[:full_path]
-    url ||= (options[:path] || send("#{auto_complete_controller_name}_path")) + "/auto_complete_#{attr}"
-    props = {
-      name: "#{f.object_name}[#{attr}]",
-      controller: options[:path] || auto_complete_controller_name,
-      url: url,
-      isDisabled: options[:disabled] || false,
-      query: f.object.search || '',
-      error: f.object.errors[attr] && f.object.errors[attr][0],
-      useKeyShortcuts: options[:use_key_shortcuts] || false,
-      id: options[:id] || "autocomplete-#{Foreman.uuid}"
-    }
-
-    container_class_name = "container-#{props[:id]}"
-    content_tag(:div, nil, :class => container_class_name) +
-      mount_react_component('FormAutocomplete', ".#{container_class_name}", props.to_json, flatten_data: true)
+    options.merge!(
+      {
+        url: options[:full_path] || (options[:path] || send("#{auto_complete_controller_name}_path")) + "/auto_complete_#{attr}",
+        controller: options[:path] || auto_complete_controller_name,
+        search_query: f.object.search || '',
+        use_key_shortcuts: options[:use_key_shortcuts] || false,
+      }
```

Comment:
Can you add more of the component props here?
For example:
- `id` which is required, 
- the `query` and `error` that can be initially shown.
- `disable`

In my opinion it will help the developers to understand all of the options this component provides,
instead of finding the react code by themselves.

Reply:
Well theese options are just those, which needs some default transformations...
There is no documentation for our `form_helper` inputs and I agree there would help one.
But I would prefer option documentation above method for that.